contrib/opentimelineio_contrib/application_plugins/rv/example_otio_reader/example_otio_reader_plugin.py

```diff
@@ -0,0 +1,168 @@
+"""
+Example plugin showing how otio files can be loaded into an RV context
+"""
+#
+# Copyright 2019 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+import os
+
+from rv import commands
+from rv import rvtypes
+
+import opentimelineio as otio
+
+import otio_reader
+
+
+class Mode(object):
+    sleeping = 1
+    loading = 2
+    processing = 3
+
+
+class ExampleOTIOReaderPlugin(rvtypes.MinorMode):
+    def __init__(self):
+        super(ExampleOTIOReaderPlugin, self).__init__()
+        self.init("example_otio_reader",
+                  [("incoming-source-path",
+                    self.incoming_source_path,
+                    "Catch otio supported files and return movieproc"),
+                   ("source-group-complete",
+                    self.source_group_complete,
+                    "Expand otio supported files (synchronous load)"),
+                   ("after-progressive-loading",
+                    self.after_progressive_loading,
+                    "Expand otio supported files (asynchronous load)")],
+                  None)
+
+        # Start as sleeping
+        self.mode = Mode.sleeping
+
+    def incoming_source_path(self, event):
+        """
+        Detects if otio file is being loaded, and replaces with
+        empty movie proc containing an otioFile tag. This will be
+        replaced in after loading.
+        """
+        event.reject()
+
+        parts = event.contents().split(';')
+        in_path = parts[0]
+        _, ext = os.path.splitext(in_path)
+        if ext:
+            ext = ext[1:]
+
+        if ext in otio.adapters.suffixes_with_defined_adapters(read=True):
```

Comment:
for cases where it isn't in the defined adapter list, maybe it would be nice to get a warning to the console?

Reply:
I worry about that actually. This code will be called for *every* file opened by RV which is often just media. It only "wakes up" when when it seems something it supports and ignores everything else (ie `.mov`, `.exr`, etc). There is a bigger question of if this is ok, or should it only wake up for `.otio` files. This above code will wake up for any format that an otio adapter supports, but might overlap with existing RV file format support.

Reply:
Ah I see.  Interesting... lets leave it as is then and see if we hit collisions (or maybe someone from autodesk could weigh in with some best practice?)